server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -122,6 +152,10 @@ protected void newResponseAsync(
         final List<FailedNodeException> failures,
         final ActionListener<ClusterStatsResponse> listener
     ) {
+        if (request.isRemoteStats()) {
```

Comment:
Could we do this after the `assert` lines which check the various preconditions of this method?